gtrak/gtreveroot.F

```diff
@@ -78,20 +78,35 @@ SUBROUTINE GTREVEROOT
       MTRACK=-999
       FINISHPRIMA=0
 10    ISPRIMA=0
-      IF(FINISHPRIMA.GE.1) THEN
+* Finish previous primary immediately
+      IF(FINISHPRIMA.GE.2) THEN
```

Comment:
The option to call FinishPrimary() immediately makes no good sense, as there is already called PostTrack()function at this phase. Instead of this, the multiple engine mode should be adapted and call FinishPrimary() after all its secondaries finished tracking.

Reply:
By default, Geant4 transports each track till the end (of this track) and then starts transporting its secondaries, the last one starts first (with the exception of using G4SmartTrackStack, which is however optional)  and each track is transported till its end before starting a new one. An exception to this behaviour are optical photons in Cerenkov and Scintillation processes, which by default transport starts before their mother track is finished. This behaviour can be changed by UI command (which we use in AliRoot productions):
`geant4->ProcessGeantCommand("/process/optical/setTrackSecondariesFirst Cerenkov 0");`

The default G4 stacking can be modified in user application; we use in AliRoot a special stacking defined in TG4SpecialStackingAction (activated via an argument of TG4RunConfiguration in g4Config.C).